### PR TITLE
Add Type Checking to ParseToken Map

### PR DESCRIPTION
**Pull request by Evanition**
Added type checking to the ParsedToken map to create a more type safe interface, also added some unit testing to verify it was working as expected

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -152,5 +152,6 @@
     "watch": "1.0.2",
     "webpack": "4.46.0",
     "yargs": "17.7.1"
-  }
+  },
+  "dependencies": {}
```
Comment (renkelvin) [inline package.json:156]: Seems not needed

File: packages/auth/package.json [hunk 2]
```diff
@@ -114,6 +114,7 @@
     "@firebase/component": "0.6.4",
     "@firebase/logger": "0.4.0",
     "@firebase/util": "1.9.3",
+    "karma-cli": "2.0.0",
```
Comment (renkelvin) [inline packages/auth/package.json:117]: Can you double check if this dependency is needed?

**Review by prameshj — APPROVED** [review]

File: package.json (hunk 1 above)
Comment (Evanition) [reply package.json]: Deleted it!

File: packages/auth/src/core/user/id_token_result.test.ts [hunk 3]
```diff
@@ -139,4 +140,15 @@ describe('core/user/id_token_result', () => {
       'Firebase: An internal AuthError has occurred. (auth/internal-error).'
     );
   });
+
+  it('Parses custom claims with multiple types', () => {
+    const token: ParsedToken = {
+      'string_claim': 'foo',
+      'object_claim': 'bar',
```
[the remark this answers is not in the document]
Comment (Evanition) [reply packages/auth/src/core/user/id_token_result.test.ts]: Go it!

File: packages/auth/package.json (hunk 2 above)
Comment (Evanition) [reply packages/auth/package.json:117]: nevermind

**Review by renkelvin — APPROVED** [review]

**Review by hsubox76 — APPROVED** [review]

**Pull request merged**